packages/flutter/lib/src/widgets/text.dart

```diff
@@ -454,6 +455,7 @@ class Text extends StatelessWidget {
   const Text(
     String this.data, {
     super.key,
+    this.selectableId,
```

Comment:
A bit worry that we add this to Text, I think having a composable solution would be better given that you also allow arbitrary type in selectedContentRange

I wonder if you can have something like a widget
```
SelectionListener(
  onSelectionChange:
  child:
) 
```
and the input is just the List<Object?> that indicate the entire content underneath and the start/end

and then one can use it like this

```
class MyState extends State<MyWidget> {
  InlineSpan mySpan = ...;
  Widget Build(context) {
    SelectionListener(
       onSelectionChange: _updateMySpan,
       child: Text.rich(mySpan), /// or complex structure as long as its content == mySpan
     )
  }
}

Reply:
This way, developer are not force to make sense of the entire selectable content in selection area, which would be tricky that we have to provide them a way to make sense of the input that passed into the onSelectionChange callback, such as selectableId.

Reply:
With this method would a user still need a way to identify the text widget under a given SelectionListener? I'm thinking with a simple structure like a column they could just iterate through the list and assume that structure matches, but not sure if that's the case for more complex structures. Or is that where they would use the `Key` as we discussed?

Reply:
My Intention was give developer a way to reduce unrelated selectable when they deal with the selection range so that they may not need a key. but I guess at some point you may still need to add the support for key.